lib/api/include/irods/ticketAPI.hpp

```diff
@@ -0,0 +1,414 @@
+            std::string_view command_modifier1;
+            std::string_view command_modifier2 = std::to_string(constraints.value_of_property);
+            std::string_view command_modifier3 = "";
+            std::string_view command_modifier4 = "";
+
+            if (constraints.value_of_property < 0) {
+                throw std::invalid_argument("Value of the property invalid");
+            }
+
+            switch (constraints.desired_property) {
+                case ticket_property::uses_count:
+                    command_modifier1 = "uses";
+                    break;
+
+                case ticket_property::write_byte:
+                    command_modifier1 = "write-byte";
+                    break;
+
+                case ticket_property::write_file:
+                    command_modifier1 = "write-file";
+                    break;
+
+                default:
+                    throw std::invalid_argument("Ticket property not given");
+            }
+            execute_ticket_operation(conn,
+                                     command,
+                                     ticket_name,
+                                     command_modifier1,
+                                     command_modifier2,
+                                     command_modifier3,
+                                     command_modifier4,
+                                     false);
+        }
+        else {
+            throw std::invalid_argument("Wrong type object given");
+        }
+    }
+    template <typename T>
+    void setTicketConstraints(RxComm& conn, const int ticket_id, const T& constraints)
+    {
+        setTicketConstraints(conn, std::to_string(ticket_id), constraints);
+    }
+
+    template <typename T>
+    void setTicketConstraints(admin_tag, RxComm& conn, const std::string_view ticket_name, const T& constraints)
+    {
+        if constexpr (std::is_same_v<ticket_property_constraint, T>) {
+            std::string_view command = "mod";
+            std::string_view command_modifier1;
+            std::string_view command_modifier2 = std::to_string(constraints.value_of_property);
+            std::string_view command_modifier3 = "";
+            std::string_view command_modifier4 = "";
+
+            if (constraints.value_of_property < 0) {
+                throw std::invalid_argument("Value of the property invalid");
+            }
+
+            switch (constraints.desired_property) {
+                case ticket_property::uses_count:
+                    command_modifier1 = "uses";
+                    break;
+
+                case ticket_property::write_byte:
+                    command_modifier1 = "write-byte";
+                    break;
+
+                case ticket_property::write_file:
+                    command_modifier1 = "write-file";
+                    break;
+
+                default:
+                    throw std::invalid_argument("Ticket property not given");
+            }
```

Comment:
This could be a function.

Reply:
Changed it to used different structs instead for each ticket property